docs/Makefile

```diff
@@ -17,4 +20,21 @@ help:
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
-	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
\ No newline at end of file
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: clean
+clean:
+	rm -rf $(BUILDDIR)/*
+
+.PHONY: html
+html:
+	# Make interface and prediction to packages so apidoc can detect them
+	touch interface/__init__.py
```

Comment:
Problem here is that if we fail to complete the build, we leave these files lying around in the repo. This is both messy, but also a problem in that even if we ``.gitignore``d them, as they are now packages this is going surely going to cause some grief wrt tests or.. something. Anyway, gut feel is that this is wrong. :/